content/trace.md

```diff
@@ -1,137 +1,134 @@
-
-If a user request is slow, how do we determine why?  
-
-* Maybe one of the two database lookups was slow. Which one?
-* Maybe the app server responded in time and the load balancer caused the slow-down.  
-
-Given a distributed trace of the slow user request, we can tell at a glance what happened.  
-&nbsp;  
+Each unit work is called a span in a trace. Spans include metadata about the work,
+including the time spent in the step (latency) and status.
+You can use tracing to debug errors and
+latency issues of your application.  
 
 ---
 
-#### Tracing Concepts  
-
-What follows is a list of all Tracing concepts and how to apply them:  
+## Spans
 
-__Trace__  
-A Trace is a collection of nested Spans. Traces often extend across multiple nodes in a distributed system. Traces are uniquely identified by a TraceId and all Spans within a Trace will have the same TraceId.  
+A trace is a tree of spans.
 
-__Span__  
-A Span represents an operation or a unit of work. Multiple spans can be part of a "Trace", which represents an execution path (usually distributed) across multiple processes/nodes. Spans are uniquely identified by a SpanId and a TraceId. Spans within the same trace have the same TraceId.  
+A span is the unit work represented in a trace. A span may
+represent a HTTP request, an RPC, a server handler,
+a database query or a section customly marked in user code.
 
-All completed Spans have: 
+![A trace](/img/trace-trace.png)
 
-* TraceId
-* SpanId
-* Start Time
-* End Time
-* Status
+Above, you see a trace with various spans. In order to respond
+to `/messages`, several other internal requests are made. First,
+we are checking if the user is authenticated, we are trying to
+get the results from the cache. It is a cache miss, hence we
+query the database for the results, we cache the results back,
+and respond back to the user.
 
-Spans can optionally have:  
+There are two types of spans:
 
-* Parent SpanId
-* Remote Parent
-* Attributes
-* Annotations
-* Message Events
-* Links
+* **Root span**: Root spans don't have a parent span. There are the
+  first span. `/messages` span above is a root span.
+* **Child span**: Child spans are a children of an existing span. 
 
-__Starting/Ending Spans__  
-When a Span starts, it records a start time.  When a Span ends, the end time is recorded. Latency is calculated by the difference between the start and end times.  Spans will only be exported to exporting services after they have ended. Attributes, annotations, message events, etc. can only be added to span that is active (i.e. has started but has not ended).  
 
-__Status__  
-Status represents the current state of the span. It is represented by a canonical status code which maps onto a predefined set of error values and an optional string message.  
+Spans are identified with an ID and are associated to a trace.
+These identifiers together are called span context.
```

Comment:
I think span context also contains trace options in addition to trace id and span id?